glossary.md

```diff
@@ -0,0 +1,64 @@
+# Glossary
+
+This document defines some terms that are used across this specification.
+
+<details>
+<summary>Table of Contents</summary>
+
+<!-- toc -->
+
+- [User Roles](#user-roles)
+  * [Application Author](#application-author)
+  * [Provider Author](#provider-author)
+  * [Integration Author](#integration-author)
+  * [Library Author](#library-author)
+- [Common](#common)
+  * [Feature Flag API](#feature-flag-api)
+  * [Feature Flag SDK](#feature-flag-sdk)
+  * [Provider](#provider)
+  * [Integration](#integration)
+  * [Context](#context)
+
+<!-- tocstop -->
+
+</details>
+
+## User Roles
+
+### Application Author
+
+The maintainer of an application or service which utilizes the feature flags API to manage functionality.
+
+### Provider Author
+
+The maintainer of an SDK-compliant provider which implements the necessary interfaces required for flag evaluation.
+
+### Integration Author
+
+The maintainer of an SDK-compliant integration which implements additional secondary functionality besides flag evaluation.
+
+### Library Author
+
+The maintainer of a shared library which is a dependency of many applications or libraries, which utilizes the feature flags API to allow consumers to manage library functionality.
+
+## Common
+
+### Feature Flag API
+
+The interfaces and abstractions used by Application Author to implement feature flags in their application or service.
+
+### Feature Flag SDK
+
+The interfaces and abstractions used by Provider Authors and Integration Authors to add support for their feature flag implementation or integration.
+
+### Provider
+
+An SDK-compliant feature flag implementation that is abstracted by the Feature Flag API. Implementations may include Saas feature flag vendors, custom "in-house" feature flag infrastructure, or open-source implementations.
```

Comment:
```suggestion
An SDK-compliant feature flag implementation which adheres to the Feature Flag API. Implementations may include Saas feature flag vendors, custom "in-house" feature flag infrastructure, or open-source implementations.
```
I had to read this like 3 times before it clicked what we were saying. This might not be the wording either, but I think we need to change a small something here.

Reply:
>SDK and API feel reversed here to me.

> API is the thing interface that we write against. We implement the interface.
SDK is the thing we use to develop the software. The "we" in this case feels more like an end user (e.g. applicatino author) to me.

Thanks for the input... This is the sort of confusion I was hoping to address. I see what you're saying. I guess it's a matter of perspective.

In my mental model, the OpenFeature API is the consistent abstraction (facade) that first party application developers need to understand to use feature flags in their software. It's intellectually lightweight. The implementation (providers) are hidden from them.

The SDK is what maintainers who would work on providers and integrations would work with - it of course also has a contract that must be fulfilled, and it's a more complex one. The larger set of more complex interfaces is why I though this is really the "SDK". I do however see your point, and I dont exactly feel strongly which direction we go in, as long as it's well defined.